music_assistant/providers/genius_lyrics/__init__.py

```diff
@@ -0,0 +1,127 @@
+
+Used for retrieval of lyrics.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from music_assistant_models.enums import ProviderFeature
+from music_assistant_models.media_items import MediaItemMetadata, Track
+
+from music_assistant.models.metadata_provider import MetadataProvider
+
+if TYPE_CHECKING:
+    from music_assistant_models.config_entries import ConfigEntry, ConfigValueType, ProviderConfig
+    from music_assistant_models.provider import ProviderManifest
+
+    from music_assistant.mass import MusicAssistant
+    from music_assistant.models import ProviderInstanceType
+
+from lyricsgenius import Genius
+
+from .helpers import clean_song_title, cleanup_lyrics
+
+SUPPORTED_FEATURES = {
+    ProviderFeature.TRACK_METADATA,
+}
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    return GeniusProvider(mass, manifest, config)
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # ruff: noqa: ARG001
+    return ()  # we do not have any config entries (yet)
+
+
+class GeniusProvider(MetadataProvider):
+    """Genius Lyrics provider for handling lyrics."""
+
+    async def handle_async_init(self) -> None:
+        """Handle async initialization of the provider."""
+        self._genius = Genius("public", skip_non_songs=True, remove_section_headers=True)
+
+    @property
+    def supported_features(self) -> set[ProviderFeature]:
+        """Return the features supported by this Provider."""
+        return SUPPORTED_FEATURES
+
+    async def get_track_metadata(self, track: Track) -> MediaItemMetadata | None:
+        """Retrieve synchronized lyrics for a track."""
+        if track.metadata and (track.metadata.lyrics or track.metadata.lrc_lyrics):
+            self.logger.debug("Skipping lyrics lookup for %s: Already has lyrics", track.name)
+            return None
+
+        if not track.artists:
+            self.logger.debug("Skipping lyrics lookup for %s: No artist information", track.name)
+            return None
+
+        artist_name = track.artists[0].name
+
+        if not track.name or len(track.name.strip()) == 0:
+            self.logger.debug(
+                "Skipping lyrics lookup for %s: No track name information", artist_name
+            )
+            return None
+
+        song_lyrics = self._fetch_lyrics(artist_name, track.name)
```

Comment:
```suggestion
        song_lyrics = await asyncio.to_thread(self._fetch_lyrics, artist_name, track.name)
```
Blocking IO must be run in an executor thread, otherwise you completely block the asyncio eventloop